CitationStyleLanguagePlugin.php

```diff
@@ -0,0 +1,1048 @@
+            $citationData->abstract = htmlspecialchars(strip_tags($publication->getLocalizedData('abstract')));
+            $citationData = $this->setArticleAuthors($citationData,$publication, $context);
+            $citationData->URL = $request->getDispatcher()->url($request, PKPApplication::ROUTE_PAGE, null, 'article', 'view', $submission->getBestId());
+            if ($publication->getDoi()) {
+                $citationData->DOI = $publication->getDoi();
+            }
+        } elseif ($this->isBook()) {
+            $citationData->type = 'book';
+            $citationData->risType = 'BOOK';
+            $citationData->id = $submission->getId();
+            $citationData->title = $publication->getLocalizedFullTitle();
+            $citationData = $this->addSeriesInformation($citationData, $publication);
+            $citationData->publisher = htmlspecialchars($context->getLocalizedName());
+            $citationData->keywords = $keywords[Locale::getLocale()] ?? [];
+            if ($publication->getData('pages')) {
+                $citationData->page = htmlspecialchars($publication->getData('pages'));
+            }
+            $citationData->abstract = htmlspecialchars(strip_tags($publication->getLocalizedData('abstract')));
+            $citationData->serialNumber = $this->getSerialNumber($publication);
+            $citationData = $this->setBookAuthors($citationData, $publication, $context);
+            $citationData->URL = $request->getDispatcher()->url($request, PKPApplication::ROUTE_PAGE, null, 'catalog', 'book', $submission->getBestId());
+            if ($publication->getDoi()) {
+                $citationData->DOI = $publication->getDoi();
+            }
+        } elseif ($this->isBookChapter()) {
+            $citationData->type = 'chapter';
+            $citationData->risType = 'CHAP';
+            $citationData->id = $chapter->getSourceChapterId();
+            $citationData->title = $chapter->getLocalizedFullTitle();
+            $citationData->{'container-title'} = $publication->getLocalizedFullTitle();
+            $citationData = $this->addSeriesInformation($citationData, $publication);
+            $citationData->publisher = htmlspecialchars($context->getLocalizedName());
+            if ($chapter->getPages()) {
+                $citationData->page = htmlspecialchars($chapter->getPages());
+            }
+            $citationData->abstract = htmlspecialchars(strip_tags($chapter->getLocalizedData('abstract')));
+            $citationData->serialNumber = $this->getSerialNumber($publication);
+            $citationData = $this->setBookChapterAuthors($citationData, $publication, $context, $chapter);
+            $citationData->URL = $request->getDispatcher()->url($request, PKPApplication::ROUTE_PAGE, null, 'catalog', 'book', [$submission->getBestId(), 'chapter', $chapter->getSourceChapterId()]);
+
+            if ($chapter->getDoi()) {
+                $citationData->DOI = $chapter->getDoi();
+            }
+        }
```

Comment:
The `else` case here should never be executed, right? If not, I'd suggest adding a
```
} else {
    throw new Exception('Unknown submission content type!');
}
```
...or equivalent.